### DISCUSSION TITLE
docs/logging-forwarder-aggregator/

### DISCUSSION THREAD

**Discussion by giscus** [ricsanfre/pi-cluster#242] [Announcements]
# docs/logging-forwarder-aggregator/

How to deploy logging collection, aggregation and distribution in our Raspberry Pi Kuberentes cluster. Deploy a forwarder/aggregator architecture using Fluentbit and Fluentd. Logs are routed to Elasticsearch and Loki, so log analysis can be done using Kibana and Grafana.

https://picluster.ricsanfre.com/docs/logging-forwarder-aggregator/

**Comment by i5Js** [comment]
Great tutorial. 

Quick question, I don't want to get this accesible from WAN. Should I need to create the certificate?

Thanks.

**Reply by ricsanfre** [reply]
Thanks @i5Js, 

It is up to you. You can decide not to secure communications between fluentbit collectors and fluentd aggregator using TLS or even not  to use authentication (shared-key). I decided to do so I can have external fluent-bit collectors, running on external hosts (in my case 'gateway' and 'pimaster' nodes), pushing logs into fluentd running within the cluster.

If you do not want to use secure connections, you only need to change a couple of lines in both fluentbit and fluentd configuration.

Regards
Ricardo

**Comment by i5Js** [comment]
You have a typo when you create the secret:

metadata: but you have metadata

**Reply by i5Js** [reply]
Another typo:

volumes:
  - name: fluentd-tls
    secret:
      secretName: fluentd-tls
     --->  - name: etcfluentd-template <---- That is duplicated, not needed
  - name: etcfluentd-template

**Reply by ricsanfre** [reply]
Thanks Javi,

Typos fixed

**Comment by i5Js** [comment]
Probably I will have another typo, but I'm getting this error in the fluentd
`2024-03-28 14:08:19 +0000 [warn]: #0 [out_es] Could not communicate to Elasticsearch, resetting connection and trying again. [400] {"error":{"root_cause":[{"type":"x_content_parse_exception","reason":"[1:12] [policy] unknown field [phases]"}],"type":"x_content_parse_exception","reason":"[1:12] [policy] unknown field [phases]"},"status":400}`

**Reply by i5Js** [reply]
Running out of ideas... Google doesn't help.

**Reply by ricsanfre** [reply]
Which ES version are you using?

**Reply by i5Js** [reply]
Same than your blog.
